app/lib/languages.rb

```diff
@@ -2,12 +2,12 @@
 
 # Helper methods for languages
 module Languages
-  def self.supported_language?(lang)
+  def self.supported_language?(lang, medium = nil)
     return false if lang.nil?
 
     # If the supported field exists at all for a language, then
     # that language can be considered supported
-    all_languages&.dig(lang.to_sym, :supported)
+    medium.present? ? all_languages&.dig(lang.to_sym, :supported, medium) : all_languages&.dig(lang.to_sym, :supported)
```

Comment:
```suggestion
    medium.present? ? all_languages&.dig(lang.to_sym, :supported, medium) : all_languages&.dig(lang.to_sym, :supported) || false
```

Also you can remove the `voice_supported?` method below this and replace its usage.